cadcdata/cadcdata/storageinv.py

```diff
@@ -606,6 +669,49 @@ def _get_md5sum(self, filename):
                 hash_md5.update(chunk)
         return hash_md5.hexdigest()
 
+    def _get_uris(self, target):
+        # takes a target URI and if the URI is not fully qualified (schema
+        # is missing, returns a list of possible fully qualified
+        # corresponding URIs
+        parts = urlparse(target)
+        if parts.scheme:
+            return [target]
+
+        scheme_file = os.path.join(
+            os.path.dirname(self._data_client.caps.caps_file),
+            '.data_uri_scheme_map')
+        scheme_url = self._data_client.caps.caps_urls[DATA_RESOURCE_ID].replace('/capabilities', '/uri-scheme-map')
+        content = self._data_client.caps._get_content(scheme_file, scheme_url)
+        schemes = self._parse_scheme_config(content)
+        archive = target.split('/')[0]
+        if archive in schemes:
+            return ['{}:{}'.format(x.strip(':'), target) for x in schemes[archive]]
+        else:
+            return ['{}:{}'.format(schemes['default'][0], target)]
+
+    def _parse_scheme_config(self, content):
```

Comment:
This is an opinion comment only, so ignore it if you wish.

The format of the schema config file is different from the format of other config files. Is there a reason it doesn't use the `Config` class/format from `cadcutils.utils.config.py`?

Reply:
That's a good point. I forgot about it to suggest it as the service format. Now it's a bit late as we want to get this out of the door.